src/main/java/com/software/modsen/rideservice/controller/RideController.java

```diff
@@ -80,9 +82,46 @@ public ResponseEntity<RideResponseSet> getRidesByStatus(@PathVariable RideStatus
         return new ResponseEntity<>(rides, HttpStatus.OK);
     }
 
-    @PatchMapping("/{id}/status")
-    public ResponseEntity<RideResponse> changeRideStatus(@PathVariable Long id, @RequestBody RideStatus rideStatus) {
-        RideResponse ride = rideMapper.rideToRideResponse(rideService.changeStatus(id, rideStatus));
+    @PutMapping("/status")
+    public ResponseEntity<RideResponse> changeRideStatus(@RequestBody RideStatusRequest rideStatus) {
+        RideResponse ride = rideMapper.rideToRideResponse(rideService.changeStatus(rideStatus));
         return new ResponseEntity<>(ride, HttpStatus.OK);
     }
+
+    @PutMapping("/driver")
+    public ResponseEntity<RideResponse> setDriver(@RequestBody RideDriverRequest request){
+        RideResponse rideResponse = rideMapper.rideToRideResponse(rideService.setDriver(request));
+        return ResponseEntity.ok(rideResponse);
+    }
+
+    @GetMapping("/created/{id}")
+    public ResponseEntity<RideResponseSet> getRideToConfirm(@PathVariable Long id){
+        return ResponseEntity.ok(new RideResponseSet(rideService.getRidesToConfirm(id)
+                .stream()
+                .map(rideMapper::rideToRideResponse)
+                .collect(Collectors.toSet())));
+    }
+
+    @PostMapping("/accept/{id}")
+    public ResponseEntity<RideResponse> acceptRide(@PathVariable Long id){
+        RideResponse rideResponse = rideMapper.rideToRideResponse(rideService.acceptRide(id));
+        return ResponseEntity.ok(rideResponse);
+    }
+
+    @PostMapping("/reject/{id}")
+    public ResponseEntity<RideResponse> rejectRide(@PathVariable Long id){
+        RideResponse rideResponse = rideMapper.rideToRideResponse(rideService.rejectRide(id));
+        return ResponseEntity.ok(rideResponse);
+    }
+
+    @PostMapping("/completed/{id}")
+    public ResponseEntity<RideResponse> completedRide(@PathVariable Long id){
+        RideResponse rideResponse = rideMapper.rideToRideResponse(rideService.completedRide(id));
+        return ResponseEntity.ok(rideResponse);
+    }
```

Comment:
It seems to me that you can use one method with specifying status in request body instead of creating three methods with logic described in the route

Reply:
changed